content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,430 @@
+
+```text
+Argument bool: Got invalid value 'true' on prisma.createOneD. Provided String, expected Boolean.
+Argument byteA: Got invalid value 'hello prisma ⚡️🚀' on prisma.createOneD. Provided String, expected Bytes.
+Argument xml: Got invalid value 123 on prisma.createOneD. Provided Int, expected String.
+```
+
+becomes the following in Prisma 5:
+
+```text
+Argument `bool`: Invalid value provided. Expected Boolean, provided String.
+```
+
+### <inlinecode>cockroachdb</inlinecode> provider is now required when connecting to a CockroachDB database
+
+### Removal of <inlinecode>runtime/index.js</inlinecode> from generated client
+
+#### Using public APIs from <inlinecode>runtime/index.js</inlinecode>
+
+Importing from `@prisma/client/runtime` is no longer available in Prisma 5. If you are using public APIs available in this namespace you can instead import `Prisma` and access them. For example:
+
+```js
+import { Decimal } from '@prisma/client/runtime'
+```
+
+will need to be changed to
+
+```js
+import { Prisma } from '@prisma/client'
+Prisma.Decimal
+```
+
+#### Using private APIs for a specific runtime
+
+Instead of importing private APIs from the top-level `@prisma/client/runtime` you will need to import from the runtime specific to your database engine. For example:
+
+```text
+@prisma/client/runtime/data-proxy
+```
+
+## Other changes
+
+The following changes may cause an application to initially throw an error message after upgrading to Prisma 5. Fortunately, they are easy to solve, as the underlying functionality has been removed for a while or the change is a simple string replace.
+
+### Removal of deprecated Prisma CLI flags
+
+Several deprecated CLI flags have been removed.
+
+- `--preview-feature` used in `db execute`, `db seed`, and `db diff`
+- `--experimental` and `--early-access-feature` used in `migrate`
+- `--force`/`-f` used in `db push`
+- `--experimental-reintrospection` and `--clean` used in `db pull`
+
+The use of `db push --force` can be replaced with the newer implementation `db push --accept-data-loss`.
+
+All other flags are from previous APIs and are no longer needed.
+
+### Removal of the <inlinecode>beforeExit</inlinecode> hook from the library engine
+
+Instead of the existing `beforeExit` hook we recommend using built-in Node.js exit events.
+
+The following code with Prisma 4:
+
+```js
+const exitHandler = () => {
+  // your exit handler code
+}
+
+prisma.$on('beforeExit', exitHandler)
+```
+
+Could become:
+
+```js
+const exitHandler = () => {
+  // your exit handler code
+}
+
+process.on('exit', exitHandler)
+process.on('beforeExit', exitHandler)
+process.on('SIGINT', exitHandler)
+process.on('SIGTERM', exitHandler)
+process.on('SIGUSR2', exitHandler)
+```
+
+### Removal of deprecated <inlinecode>prisma2</inlinecode> executable
+
+When Prisma 2 was released a new executable, `prisma2` was added to the CLI in addition to the existing `prisma` executable. Today we are removing the extra executable.
```

Comment:
Correct story is: When we released Prisma 2, many people were still using Prisma 1 which also was using `prisma`. So the first few releases actually only had `prisma2`, before we later added `prisma` and renamed the PRisma 1 one to `prisma1`.

Not sure if this level of info is relevant. You could also just say that "Prisma CLI used to have a `prisma2` executable, which we deprecated a long time ago and now removed it".

Reply:
Thanks for the info! I took that and added some color to it.